dashboard/src/App.js

```diff
@@ -1,24 +1,17 @@
-import React, { useEffect } from 'react';
-import favicon from './assets/logo/favicon.ico';
-import './App.css';
-import MainLayout from "./modules/containers/MainLayout/index";
-import '@patternfly/patternfly/patternfly.css';
+import React from 'react'
+import { BrowserRouter as Router,Route,Routes } from 'react-router-dom'
+import { TableWithFavorite } from './modules/components/TableComponent'
 
-
-function App() {  
-  
-  useEffect(() => {
-    const faviconLogo = document.getElementById("favicon");
-    faviconLogo.setAttribute("href", favicon);
-  }, [])
-
- 
+function App() {
   return (
-    <div className="App">
-      Pbench Dashboard
-      <MainLayout />
-    </div> 
-  );
+    <>
```

Comment:
It is a React Fragment.So, a common practice in React is a component will return multiple component.But one of shortcoming (I dont know if it should be called a short coming or limitation) of React is any JSX Expression can only have one Parent Element only and that parent component can have multiple child elements and components.So ,if we do something like this:-
`
function App()
{
   return(
       <A/>
      <B/>
      <C/>
    )
}
`
This is illegal in react and cannot be done since there is no parent element and we are trying to return 3 components from a single component.

To fix this,we can wrap these 3 components inside a `<div>` tag

Though wrapping in div tag is fine in some cases,it has a drawback.We are unnecessary introducing a extra node in our DOM Tree,which in some case can also cause trouble.

For example ,lets say we have something like this:-

`
function App(){
<table>
<tr>
<Columns/>
</tr>
</table>
}
`
The Column component will return <td> elements representing row data.Since there will be multiple <td> elements even in a basic table ,we have to wrap them inside a <div> tag.So the rendered HTML for <table> will look something like this

```
function App()
{
<table>
<tr>
<div>
<td>
<td>
</div>
</tr>
</table>
}
```

But this is a very bad HTML (div inside a tr tag),hence React Fragment comes to Picture.They are basically blank HTML tags ,and are only used as Parent element  to return multiple components from inside a React component.They do not add any extra Node in our DOM Tree and hence we get a clean and valid HTML.

Reply:
Thanks for the explanation!

It looks to me like the `<Route .../>` tag is contained within the `<Routes>` tag which is contained within the `<Router>` tag...which makes me wonder why the last needs to be wrapped in another tag...is it because `<Router>` is going to expand to more than one (non-nested) tag?

Reply:
Yes,I should probably remove it,I usually wrap the elements inside React Fragments since they are harmless ,but here I think that `<Router>` is not going to expand to more than one non-nested tag anytime soon.